src/Raven.Server/ServerWide/ServerStore.cs

```diff
@@ -2983,8 +2983,10 @@ private async Task<(long Index, object Result)> SendToNodeAsync(TransactionOpera
                 || serverCertificateChanged
                 || _clusterRequestExecutor.Url.Equals(leaderUrl, StringComparison.OrdinalIgnoreCase) == false)
             {
-                _clusterRequestExecutor?.Dispose();
-                _clusterRequestExecutor = CreateNewClusterRequestExecutor(leaderUrl);
+                var newExecutor = CreateNewClusterRequestExecutor(leaderUrl);
+                var oldExecutor = Interlocked.Exchange(ref _clusterRequestExecutor, newExecutor);
+
+                oldExecutor?.Dispose();
```

Comment:
What happens if another thread is making use of the old executer while you dispose it? Shouldn't we worry about use after free here?

Reply:
Oren, indeed, I need to ensure that no other threads are accessing the old executor before I dispose of it. I can implement a reference counting mechanism, incrementing the reference count when a thread begins using the executor and decrementing it once the thread has finished. The old executor would only be disposed of when the reference count reaches zero. Is this approach appropriate?

Reply:
Better to handle this in destructor?